viewing_party/party.py

```diff
@@ -1,23 +1,127 @@
 # ------------- WAVE 1 --------------------
 
+from typing import Counter
+
+
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+    if bool(title) == True and bool(genre) == True and bool(rating) == True:
+        return new_movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"] = [movie]
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    
+    ratings = [movie["rating"] for movie in user_data["watched"]]
+    Sum = sum(ratings)
+    Count = len(ratings)
 
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    return Sum/Count
+    
+def get_most_watched_genre(user_data):
+    most_watched = ""
+    max_value = 0
+    genres = [movie["genre"] for movie in user_data["watched"]]
+    genres_dict = Counter(genres)
 
+    if len(user_data["watched"]) == 0:
+        return None
+    for key, value in genres_dict.items():
+        if value >= max_value:
+            max_value = value
+            most_watched = key
+    return most_watched
+    
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    unique_movies = []
+    friends_titles = []
+    for friend in user_data["friends"]:
+        for i in range(len(friend["watched"])):
+            friends_titles.append(friend["watched"][i]["title"])
+            
+    friends_titles_set= set(friends_titles)
+
+    for movie in user_data["watched"]:
+        if not movie["title"] in friends_titles_set:
+            unique_movies.append(movie)
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    friends_movies = []
+    friends_unique_watched = []
+    for friend in user_data["friends"]:
+            friends_movies.extend(friend["watched"])

```

Comment:
Style: over-indented code:

```suggestion
    for friend in user_data["friends"]:
        friends_movies.extend(friend["watched"])

```